spec/features/pilot/stage_3_supply_registration_identifier_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Feature: Stage 3: Supply "Spacecraft Registration Identifier"
+#   So that my "reg" can be checked by "the authorities"
+#   As a pilot
+#   I want to supply a validated "Spacecraft Registration Identifier"
+
+RSpec.feature "Stage 3: Supply registration identifier" do
+  # Scenario: Must supply Spacecraft Registration Identifier
+  #   Given I am on the 'registration identifier' stage
```

Comment:
Is using comments for Cucumber specs a common pattern? I've seen it on a few projects, but would have thought that the verbosity of the method names below would hopefully remove the need for additional comments. More of a genuine question than something to change!

Reply:
The technique which I try to use and to encourage is to:

1. Write a BDD-style acceptance/feature/end-to-end test in the user story /  Trello / Jira ticket
2. Copy this word for word into the high-level testing framework of use, e.g. Cucumber, Rspec-Capybara, Cypress

In the case of Cucumber, the plain English spec becomes  the executable test file. In other frameworks like Rspec/Capybara (in use here) we need to turn our lines of `given/when/then/and` statements into method calls and write the matching methods.

I think you're right that once the method-calls and methods are implemented, the original comments in plain English become largely redundant and a source of maintenance error. i.e. they can become out of sync with the methods themselves. 

You don't have this issue if you use Cucumber, which I think is the best approach. I believe we ought to make Cucumber the default choice of high-level testing framework.

Focussing on the high level benefits and functionality of features is critical to delivering software which is user-centric. Cucumber is in my view the best tool to drive this practice.

Reply:
I've removed the comments from the scenarios and left the "feature" comment:

```
# Feature: Stage 3: Supply "Spacecraft Registration Identifier"
#   So that my "reg" can be checked by "the authorities"
#   As a pilot
#   I want to supply a validated "Spacecraft Registration Identifier"
```
as that does have enduring value to state the purpose of the feature.